eclair-core/src/main/scala/fr/acinq/eclair/blockchain/OnChainWallet.scala

```diff
@@ -125,4 +125,30 @@ object OnChainWallet {
   final case class FundTransactionResponse(tx: Transaction, fee: Satoshi, changePosition: Option[Int]) {
     val amountIn: Satoshi = fee + tx.txOut.map(_.amount).sum
   }
+
+  final case class ProcessPsbtResponse(psbt: Psbt, complete: Boolean) {
```

Comment:
Because otherwise the dependency was wrong: `BitcoinCoreClient` extends `OnChainWallet`, but `OnChainWallet` defined functions that returned objects that were defined in `BitcoinCoreClient`...

The functions exposed in a trait should generally not return types that are only defined in one of the child classes implementing that trait, right?

Reply:
Right, and it's consistent with the definition of `FundTransactionResponse` for example.